File: test/examples/cpp14/compute_pi.cpp

#include <cmath>

double compute_pi(int segments) {
  double polygon_edge_length_squared = double(2.0);
  int polygon_sides = 2;
  for(int _ = 0; _ < segments; ++_) {
    polygon_edge_length_squared = -sqrt(1 - polygon_edge_length_squared / 4) * 2 + 2;
    polygon_sides *= 2;
  }
  return sqrt(polygon_edge_length_squared) * polygon_sides;
}

File: test/examples/f95/compute_pi.f90

function compute_pi(segments)
  implicit none

  integer, intent(in) :: segments
  real*8 :: compute_pi

  real*8 :: polygon_edge_length_squared
  integer :: polygon_sides
  integer :: i

  polygon_edge_length_squared = 2.0
  polygon_sides = 2
  do i = 1, segments
    polygon_edge_length_squared = -sqrt(1 - polygon_edge_length_squared / 4) * 2 + 2
    polygon_sides = polygon_sides * 2
  end do
  compute_pi = sqrt(polygon_edge_length_squared) * polygon_sides
  return
end function
